main/src/main/java/cgeo/geocaching/connector/al/ALConnector.java

```diff
@@ -108,7 +108,7 @@ public String getExtraDescription() {
 
     @Override
     public boolean supportsSettingFoundState() {
-        return true;
+        return false;
```

Comment:
Thanks for working on it!

You may want to remove the method completely, as the default implementation in abstract connector also just returns false.

Reply:
I thought about it but decided to leave it in, in case at some point our API methods being used don't give us the required properties anymore. In that case we could just toggle the switch.

If you want me to take it out, I'll do of course.

Reply:
No clear preference from my side, both is fine for me. I'll wait with merging though, to give others time to comment...

Reply:
thought about it some more ... I think taking it out is ok as it can be brought back, if needed, within minutes.

Reply:
At first I thought, how does she do this  as it is not possible in cgeo to use different accounts for GC and LC but I believe I understand the use  case now. Let me see whether I got it right. You use user X for GC and LC enabled with cgeo. You use a user Y with GC's labs app to actually "play" the labs and their found state is correctly reflected in GCs app, however for your user Y. Now when you work with cgeo in the context of user X, you want to manually set the LCs you played for user Y with the app to a `found it`. Right?

I know of others who do the same. Primary reason being to not "pollute" their stats on GC with the massive amounts of "points" that accumulate by playing labs. Frequently they use basic accounts for playing the labs as there are almost no restrictions for basic users when playing labs with GC's app.

Having said all this, there is no need to change the existing functionality in cgeo. We could, if we wanted to, make this variable a configurable option, perhaps somewhere in advanced settings? On the other hand we're having discussions that there are already way too many setting switches ... so adding even more is perhaps not such a good idea either.

Reply:
> At first I thought, how does she do this  as it is not possible in cgeo to use different accounts for GC and LC but I believe I understand the use  case now. Let me see whether I got it right. You use user X for GC and LC enabled with cgeo. You use a user Y with GC's labs app to actually "play" the labs and their found state is correctly reflected in GCs app, however for your user Y. Now when you work with cgeo in the context of user X, you want to manually set the LCs you played for user Y with the app to a `found it`. Right?
> 
> I know of others who do the same. Primary reason being to not "pollute" their stats on GC with the massive amounts of "points" that accumulate by playing labs. Frequently they use basic accounts for playing the labs as there are almost no restrictions for basic users when playing labs with GC's app.
> 

You got it right, that's exactly how and why I do this

Reply:
> Having said all this, there is no need to change the existing functionality in cgeo. We could, if we wanted to, make this variable a configurable option, perhaps somewhere in advanced settings? On the other hand we're having discussions that there are already way too many setting switches ... so adding even more is perhaps not such a good idea either.

I will be fine with that setting, but would prefer the possibility to use a different user for the labs in c:geo - but I guess that would probably not be so easy

Reply:
I'm one of the guys trying to stop every new setting. ;) 

However I'm primarily against settings which could be avoided by just finding good defaults and build better UIs. In this case it definitely makes sense to make it configurable, and the LC connector specific settings page is still pretty clean so we could add it there.

![Screenshot_20240202_104214_cgeo geocaching_copy_270x585](https://github.com/cgeo/cgeo/assets/64581222/b435fd49-fabf-4528-b401-bf8a2aef2e39)

Reply:
So the option would be something like "I use a different gc-account" or " don't fetch found state from gc-account" or "..."
I am not sure which wording we should use

Reply:
> would prefer the possibility to use a different user for the labs in c:geo

I'm not sure, can basic members access the API methods we use at all? And what would GC do if they recognize that we seem to let basic members use lab methods (which we currently don't) even if it's just the secondary account of a PM?

Reply:
> > would prefer the possibility to use a different user for the labs in c:geo
> 
> I'm not sure, can basic members access the API methods we use at all? And what would GC do if they recognize that we seem to let basic members use lab methods (which we currently don't) even if it's just the secondary account of a PM? 

Good point - as I mentioned before - I can live with the setting then

Reply:
Technically wise, removing the method from `ALConnector` would be the way to go, as @fm-sys has mentioned above.

With regard to @murggel's use-case, I propose something different, though:
- Open a separate issue for optionally using `ALConnector` with a separate user,
- and wait with removing the option to manually set found state for `ALConnector` until that feature request is implemented.

Reply:
> > would prefer the possibility to use a different user for the labs in c:geo
> 
> I'm not sure, can basic members access the API methods we use at all? And what would GC do if they recognize that we seem to let basic members use lab methods (which we currently don't) even if it's just the secondary account of a PM?

To all I know, technically the APIs we're consuming also work for basic accounts exactly the way they do for premium.

The LC related API methods in the partner API, aka GC Live API are scoped at premium accounts only. Since we aren't using those we have technically no restrictions. Just like GC's Lab App, which also does not restrict basic accounts by any means. With one exception: There seem to be some recent functions in the GC Lab App which also require premium but to all I know these are more or less irrelevant.

Scoping our implementation at premium account is purely to be good citizen and to mimic the behaviour an "API partner app" would be forced to.

Reply:
> Technically wise, removing the method from `ALConnector` would be the way to go, as @fm-sys has mentioned above.
> 
> With regard to @murggel's use-case, I propose something different, though:
> 
> * Open a separate issue for optionally using `ALConnector` with a separate user,
> * and wait with removing the option to manually set found state for `ALConnector` until that feature request is implemented.

I think her use case wouldn't work anymore as her "user Y" (as I called it), would certainly be a basic user.  In order to make this useful, we would have to remove some voluntary restrictions from our implementation. Which pretty much relates to the quesiton @fm-sys asked further up in this thread.

Reply:
Let me add some clarification with respect to what "a user" is in the context of the AL connector: When it comes to comsuming the labs api, we are not in a logged in state, we do not have any `access_token` for a specific user. Neither today nor in the forseeable future. Most of the AL functionality in cgeo is completely independet from a GC account, except **one** subtle association and that is the one we need in order to determine the true found state of an AL. In order to retrieve that all we need to know is the `public GUID` but no authorization at all. I would consider this a "side effect" of that API method, perhaps, in the light of CCPA an even unwanted side effect, a side effect which may go away the more we talk about it the more likely.

So if we provided a way for a user to manually enter the public GUID of **any** user we would get the found state for that user. Let's link that back to the "user Y" in the example further above, the use case we discussed would be working.

Also, and to be complete: Our `consumer_key` does not support to "login" a user, at least not with the now deprecated `login` method, perhaps it would work via OAuth2, I haven't tested that but I wouldn't be surprised if that didn't work either as we are not supposed to really have a logged in context, in which, if we had it, we could actually "solve" those labs by submitting correct answers.

With all that no clarified, @moving-bits, would it make sense to extend the AL connector to have some sort of a login to a specific account? We could sort of mimic a login to that user's GC account but only in order to retrieve that `public-guid` and nothing else.

Reply:
Thanks @baiti for the additional details. We could then implement something like this:
- checkbox for AL connector "use GC account data" (default = true)
- if unchecked: show credentials form for username/password
  - if data in this fields get changed: login with those credentials, and on successful login store guid for that "AL user" in our settings
- when doing requests against the API, either use guid for "GC user" or from "AL user", depending on setting
- don't refresh AL guid on every request, but only when credentials are changed (or when switching "use GC account data" to false)
- I'm not sure if we can reuse the existing GC login methods (but we should try to) - we need to check for possible side-effects

Reply:
My few cents (I hope I did not miss that part of the discussion):
Do we need to consider basic members who import lab caches via GPX? Those users should still be able to set the found state locally. Am I right?

Reply:
> Thanks @baiti for the additional details. We could then implement something like this:
> 
> * checkbox for AL connector "use GC account data" (default = true)
> * if unchecked: show credentials form for username/password
>   
>   * if data in this fields get changed: login with those credentials, and on successful login store guid for that "AL user" in our settings
> * when doing requests against the API, either use guid for "GC user" or from "AL user", depending on setting
> * don't refresh AL guid on every request, but only when credentials are changed (or when switching "use GC account data" to false)
> * I'm not sure if we can reuse the existing GC login methods (but we should try to) - we need to check for possible side-effects

Yes, that would be a solution that support the scenario we're discussing here. However, assuming you all understand the scenario @murggel described, the account logging into the AL connector would very much likely be a basic account. Would that render sufficient functionality? @murggel, could you do a quick test: Logout from your premium account, loginto you "user Y" account, check whether you get the necessary functionality for user Y's AL caches. I would assume it would be sufficient for you then if you can see the dots on the map and the true found state on them. You wouldn't get much more unless we changed some of our policies which we most likely wouldn't (e.g. limiting functionality to premium accounts even though we could technically do the same for basic what we do for premium).

Reply:
I will do a test this evening with my phone where I have them stored and with another one where they are not yet stored

Reply:
> I will do a test this evening with my phone where I have them stored and with another one where they are not yet stored

@baiti But since a premium account for ALC is necessary - I will only see the already stored information  - correct?

Reply:
I am not exactly sure how much you can see with a basic account.

Reply:
I would assert: if you log into your "user Y" and you don't see any AL with it, the your use case is broken and you most likely would want to continue to use your current workflow with manually setting your found state? If that is true, it would also be true for most other cachers interested in the "user Y" approach, in essence, nobody would ever log into two accounts, e.g. user X (premium) for regular GC caches and user Y (basic) for LC objects.

Reply:
But then the option "don't fetch the real found state" would make sense for my workflow

Reply:
I just tried it with a fresh installation  - as expected - I don't see any lab caches with my basic account
[]()![Screenshot_20240203-200206_cgeo.jpg](https://github.com/cgeo/cgeo/assets/70113341/f1b5600f-db88-47f6-9d36-d7cb6d63618a)

![Screenshot_20240203-200232_cgeo.jpg](https://github.com/cgeo/cgeo/assets/70113341/5d00d0fb-0db7-4fd2-b386-0688781615ca)